### ISSUE TITLE
MenuBar is opened by call OpenMenu even it's disabled.

### ISSUE THREAD

**Issue by BDisp** [gui-cs/Terminal.Gui#3219]
**Describe the bug**
If a `MenuBar` is disabled it can't be activated either by mouse, keyboard, method or anything that can open a menu bar.

**To Reproduce**
1. Go to 'MenuBar' scenario
2. Click on 'Toggle Menu.Enable' button
3. Click on 'Open Menu' button
4. `MenuBar` is opened and it shouldn't

**Expected behavior**
`MenuBar` only must could open if it's enabled, right?

**Comment by dodexahedron** [comment]
I think this is fixed in some of the preliminary work I've done on reworking events.

I'd suggest waiting on this one until that work can move forward, to avoid duplicating effort there.

**Comment by BDisp** [comment]
My fix is in the `OpenMenu` method. Do you remember how you fixed it? Was an issue opened for this?

```cs
	/// <summary>
	///         Opens the Menu programatically, as though the F9 key were pressed.
	/// </summary>
	public void OpenMenu ()
	{
		var mbar = GetMouseGrabViewInstance (this);
		if (mbar != null) {
			mbar.CleanUp ();
		}

		if (!Enabled || _openMenu != null) {
			return;
		}

		_selected = 0;
		SetNeedsDisplay ();

		_previousFocused = SuperView == null ? Application.Current.Focused : SuperView.Focused;
		OpenMenu (_selected);
		if (!SelectEnabledItem (openCurrentMenu.BarItems.Children, openCurrentMenu._currentChild,
			    out openCurrentMenu._currentChild) && !CloseMenu (false)) {
			return;
		}

		if (!openCurrentMenu.CheckSubMenu ()) {
			return;
		}

		Application.GrabMouse (this);
	}

```

**Comment by dodexahedron** [comment]
The thing about the work on the events is that it touches almost everything. So, as a side effect of that, methods like that one end up getting touched, usually in simple ways like that.

I don't know if there's ever been an issue for that specific problem. 🤷‍♂️

Since the event overhaul is global, though, I didn't really have any intention of filing issues for each class or method affected, because it would be over 100 issues if I did. 🤷‍♂️

#3209 is the master issue for that, but is mostly stalled until we get #3212 finished up.

**Comment by BDisp** [comment]
I will commit it only to being possible I finish the PR #3181.

**Comment by dodexahedron** [comment]
Coolio. Even if there's overlap, it's not a problem. I'll deal with conflicts as the happen. Just a heads up to save you time if you come across any others like that, as well.

**Comment by BDisp** [comment]
Closing this because it was already merged onto #3181 and #3192.
